pkg/plugins/golang/v4/scaffolds/internal/templates/makefile.go

```diff
@@ -219,31 +211,48 @@ $(LOCALBIN):
 
 ## Tool Binaries
 KUBECTL ?= kubectl
-KUSTOMIZE ?= $(LOCALBIN)/kustomize
-CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen
-ENVTEST ?= $(LOCALBIN)/setup-envtest
+KUSTOMIZE ?= $(LOCALBIN)/kustomize-$(KUSTOMIZE_VERSION)
+CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen-$(CONTROLLER_TOOLS_VERSION)
+ENVTEST ?= $(LOCALBIN)/setup-envtest-$(ENVTEST_VERSION)
+GOLANGCI_LINT = $(LOCALBIN)/golangci-lint-$(GOLANGCI_LINT_VERSION)
 
 ## Tool Versions
 KUSTOMIZE_VERSION ?= {{ .KustomizeVersion }}
 CONTROLLER_TOOLS_VERSION ?= {{ .ControllerToolsVersion }}
+ENVTEST_VERSION ?= release-0.16 # branch name
```

Comment:
What about we try to use here `ControllerRuntimeVersion` and remove the value after the first dot?
See that when we bump the controllerRuntime version we bump it here: https://github.com/varshaprasad96/kubebuilder/blob/e349099a9667e717e9be36d551c460c9a9ddfbe4/pkg/plugins/golang/v4/scaffolds/init.go#L32-L40

Otherwise, we might forget to bump the Makefile as well.

Reply:
that would be nice! basically use tag instead of branch (did not try if works :grimacing:)

Reply:
if i remember correctly, there was problem with tags. i will check it.

Reply:
go install sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.16.3                                 
go: sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.16.3: module sigs.k8s.io/controller-runtime@v0.16.3 found, but does not contain package sigs.k8s.io/controller-runtime/tools/setup-envtest

Problem is controller-runtime repo. It is multimodules repository and they don't use proper versioning for submodules.

We can technically generate branch name from version because they use the same format or we can translate tag to commit hash.

Reply:
i created issue for them: https://github.com/kubernetes-sigs/controller-runtime/issues/2646

fix isn't so hard, if they want to keep the same version for controller-runtime and setup-envtest. They just have to create one more specific tag.

Reply:
@camilamacedo86 i think that discussion about setup-envtest will take long time and i don't like long-life PR. I suggest create separate issue for it and i can prepare functionality for generate branch name from ControllerRuntimeVersion as workaround.

Do you agree with my suggestion?

Reply:
Hi @lukas016 

I think we can achive that in a very simple one. Follow the code/draft: 

Add the func: 

```go
func getControllerRuntimeReleaseBranch(version string) string {
	// Remove the "v" prefix
	version = strings.TrimPrefix(version, "v")

	// Split the version string into its components
	parts := strings.Split(version, ".")

	// Check if we have at least two components (major and minor)
	if len(parts) < 2 {
		return "release-" + version // Return the whole version if it's not in 'major.minor.patch' format
	}

	// Return the formatted version
	return "release-" + parts[0]
}
```

- Then, we add the variable after `// ControllerRuntimeVersion version to be used to download the envtest setup script
	ControllerRuntimeVersion string`

```go
	// ControllerRuntimeReleaseBranchName store the name of the branch to be used to setup ENV TEST
	ControllerRuntimeReleaseBranchName string
```

- Populate the new variable, call after:

```
if f.Image == "" {
		f.Image = "controller:latest"
	}
```

```go
f len(f.ControllerRuntimeReleaseBranchName) == 0 {
		f.ControllerRuntimeReleaseBranchName = getControllerRuntimeReleaseBranch(f.ControllerRuntimeVersion)
	}

```

- And by last, just use it in the template {{ .ControllerRuntimeReleaseBranchName }}

WDYT?

Reply:
it looks fine as workaround.

Reply:
That would make simple we keep it maintained
Also, please squash the commits so we can get this one merged. 

Thank you a lot for your contribution 🥇

Reply:
fixed